### ISSUE TITLE
4.9.1 regression on array_splice, now has extra spaces

### ISSUE THREAD

**Issue by TomasVotruba** [nikic/PHP-Parser#714]
In 4.9.0 this worked, in 4.9.1 ([their commit diff](https://github.com/nikic/PHP-Parser/compare/v4.9.0...v4.9.1)) 

Probably related to these 2 commits:

- https://github.com/nikic/PHP-Parser/commit/ad365b1beb4d12d78c0eed489b070a83c43b854f
- https://github.com/nikic/PHP-Parser/commit/bd722809f73f84a8025ba368ae2fd056f562a6b7

I've narrow it down to [the line in Rector](https://github.com/rectorphp/rector/blob/d9dd72cd6c3e64aed8062cd6bb38267c64bc6ecd/src/PhpParser/Node/Manipulator/ClassInsertManipulator.php#L93), where `array_splice` is adding stmts to `ClassLike` or `FunctionLike`. 

```php
 /**
     * @param Stmt[] $nodes
     * @return Stmt[]
     */
    public function insertBefore(array $nodes, Stmt $stmt, int $key): array
    {
        array_splice($nodes, $key, 0, [$stmt]);

        return $nodes;
    }
```

Per each added node, it nows add 1 newline above the content:

```diff
     public function go()
     {
+        
+
         $maybe = 'yes';
```

**Comment by nikic** [comment]
Can you link to the test case this affects?

**Comment by TomasVotruba** [comment]
Sure, I've update my upgrade PR to show only stmts spaces:
https://github.com/rectorphp/rector/pull/4277

**Comment by TomasVotruba** [comment]
Just a sec, need to rebase

**Comment by TomasVotruba** [comment]
This job shows it: https://github.com/rectorphp/rector/pull/4277/checks?check_run_id=1151120962

**Comment by TomasVotruba** [comment]
Another clue that might help, is when class elements are re-ordered (e.g. sorted by visibility from public, protected and private) and they have annotations.

https://github.com/rectorphp/rector/pull/4277/checks?check_run_id=1151120962#step:7:347

The annotation of topmost class element is removed:

![image](https://user-images.githubusercontent.com/924196/93924406-fcc28080-fd14-11ea-95b3-197b49daddd4.png)

**Comment by nikic** [comment]
Thanks! I have this test case now:

```
<?php
function test() {
    if ($x) {
        $a;
        $b;
    }
    $z;
}
-----
$fnStmts =& $stmts[0]->stmts;
array_splice($fnStmts, 0, 1, $fnStmts[0]->stmts);
-----
<?php
function test() {

    $a;
    $b;
    $z;
}
```

**Comment by nikic** [comment]
I applied a fix in https://github.com/nikic/PHP-Parser/commit/88be6127fa91145ebf4f02a75e4e813f09d7c99c. However, it might have broken other things :) Can you try it before I create a release?

**Comment by TomasVotruba** [comment]
I've switched to `master` and after few fixes, all tests pass: https://github.com/rectorphp/rector/pull/4277

No new bugs :) 

Thanks :+1:

**Comment by nikic** [comment]
Thanks for checking! I've now released 4.10.1 with the fix for this issue.

**Comment by TomasVotruba** [comment]
Thank you, that was fast
